napari/conftest.py

```diff
@@ -759,15 +879,121 @@ def my_start(self):
     )
 
 
+with contextlib.suppress(ImportError):
+    # in headless test suite we don't have Qt bindings
+    # So we cannot inherit from QtBot and declare the fixture
+
+    from pytestqt.qtbot import QtBot
+
+    class OwnQtBot(QtBot):
+        """This is a modified version of QtBot that
+        ensures that all widgets added to it
+        has, after the test ends, objectName set to 'handled_widget'
+        This allows distinguishing them from non-handled widgets
+        in the _find_dangling_widgets fixture.
+        """
+
+        def addWidget(self, widget, *, before_close_func=None):
+            if widget.objectName() == '':
+                # objet do not have name, so we can set it
+                widget.setObjectName('handled_widget')
+                before_close_func_ = before_close_func
+            elif before_close_func is None:
+                # there is no custom teardown function
+                # so we could provide one that will set object name
+
+                def before_close_func_(w):
+                    w.setObjectName('handled_widget')
+            else:
+                # user provided custom teardown function
+                # so we need to wrap it to set object name
+
+                def before_close_func_(w):
+                    before_close_func(w)
+                    w.setObjectName('handled_widget')
+
+            super().addWidget(widget, before_close_func=before_close_func_)
+
+    @pytest.fixture
+    def qtbot(qapp, request):  # pragma: no cover
+        """
+        Fixture used to create a OwnQtBot instance for using during testing.
+
+        Make sure to call addWidget for each top-level widget you create to ensure
+        that they are properly closed after the test ends.
+
+        The `qapp` fixture is used to ensure that the QApplication is created before,
+        so we need it, even without using it directly in this fixture
+        """
+        return OwnQtBot(request)
+
+
+@pytest.fixture
+def _find_dangling_widgets(request, qtbot):
+    yield
+
+    from qtpy.QtWidgets import QApplication
+
+    from napari._qt.qt_main_window import _QtMainWindow
+
+    top_level_widgets = QApplication.topLevelWidgets()
+
+    viewer_weak_set = getattr(request.node, '_viewer_weak_set', set())
+
+    problematic_widgets = []
+
+    for widget in top_level_widgets:
+        if widget.parent() is not None:
+            continue
+        if (
+            isinstance(widget, _QtMainWindow)
+            and widget._qt_viewer.viewer in viewer_weak_set
+        ):
+            continue
+
+        if widget.__class__.__module__.startswith('qtconsole'):
+            continue
+
+        if widget.objectName() == 'handled_widget':
+            continue
+
+        if widget.__class__.__name__ == 'CanvasBackendDesktop':
+            # leaking in the tests_sys_info.py
+            continue
```

Comment:
Is this a TODO?

Reply:
It is the same note lake existing from long time in `make_napari_viewer`. There is a problem of leaking `CanvasBackendDesktop`. I have found that in my case it happens in `test_sys_info.py` 
So we do not report it as a test failure, but I left a comment to simplify the life of someone who would like to solve this, to simplify his life.

Reply:
So you are saying, this class leaks and we can't/don't know how to fix it, so we ignore it here?

Reply:
Yes. From long time :(